integrated_channels/canvas/utils.py

```diff
@@ -0,0 +1,173 @@
+'''Collection of static util methods for various Canvas operations'''
+import logging
+from http import HTTPStatus
+
+from requests.utils import quote
+
+from integrated_channels.exceptions import ClientError
+from integrated_channels.utils import generate_formatted_log
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CanvasUtil:
+    """
+    A util to make various util functions related to Canvas easier and co-located.
```

Comment:
Sorry, what's the point of having a static object, versus this simply being a library of functions?  It's not like we have to have objects for everything in Python-land.

Reply:
yeah good point, I had started off as a regular class with non static methods (since I had thought to do the fetch of root acount in the constructor and hold on to it), but for simplicity for now, I decided to go stateless, and I ended up with all static methods, and never thought to switch away from the class.

Reply:
ever better, (in another PR), I think we want to use urllib.parse. 

```
>>> import urllib.parse
>>> urllib.parse.urljoin('http://test1','/abc')
'http://test1/abc'
>>> urllib.parse.urljoin('http://test1/','/abc')
'http://test1/abc'
>>> urllib.parse.urljoin('http://test1/','abc')
'http://test1/abc'
>>> urllib.parse.urljoin('http://test1','abc')
'http://test1/abc'
```